### PR TITLE
[kubetest2-gke deployer] Automatically resolves the release channel if it's not specified

### PR DESCRIPTION
**Pull request by chizhg**
Sometimes when the users create the GKE clusters, they just want to specify a valid cluster version but do not want to hard code the channel, but if the version is only valid in a specific channel (e.g. rapid), the cluster creation will fail.

This PR adds the support to automatically resolve the release channel if it's not specified, so that as long as the specified version is in any of the valid release channels, `kubetest2 gke --up` will work.

/cc @amwat

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -13,7 +13,9 @@ require (
 	github.com/spf13/pflag v1.0.5
 	golang.org/x/sync v0.0.0-20201207232520-09787c993a3a
 	golang.org/x/sys v0.0.0-20201221093633-bc327ba9c2f0 // indirect
+	google.golang.org/api v0.36.0
 	gopkg.in/yaml.v2 v2.4.0
+	k8s.io/apimachinery v0.18.8
```
Comment (amwat) [inline go.mod:18]: Let's avoid adding k8s.io dependencies as much as possible, especially since we need it only seem to be adding this for using sets.

File: kubetest2-gke/deployer/version.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+		}
+		if !re.MatchString(version) {
+			return fmt.Errorf("unknown version %q", version)
+		}
+	}
+	return nil
+}
+
+func validateReleaseChannel(releaseChannel string) error {
+	if releaseChannel != "" && !validReleaseChannels.Has(releaseChannel) {
```
Comment (amwat) [inline kubetest2-gke/deployer/version.go:56]: maybe can reuse `toReleaseChannel` to avoid needing sets.

File: kubetest2-gke/deployer/up.go [hunk 3]
```diff
@@ -164,6 +163,11 @@ func (d *Deployer) CreateCluster(project string, cluster cluster, subNetworkArgs
 		}
 	} else {
 		args = append(args, "--cluster-version="+d.Version)
+		releaseChannel, err := resolveReleaseChannelForClusterVersion(d.Version, locationArg)
+		if err != nil {
+			return err
```
Comment (amwat) [inline kubetest2-gke/deployer/up.go:168]: /hold

This should be a warning, since we also may want to test bleeding edge versions which aren't in any channel yet.

**Review by amwat — COMMENTED** [review]
Let's make this best-effort instead of a run failing error.
Otherwise lgtm.

Isn't None channel usually a superset of all channels except maybe Rapid?

File: go.mod (hunk 1 above)
Comment (chizhg) [reply go.mod]: Done.

File: kubetest2-gke/deployer/version.go (hunk 2 above)
Comment (chizhg) [reply kubetest2-gke/deployer/version.go]: `toReleaseChannel` is used to convert the channel name in the server config to the channel name in the `gcloud` flag, so it cannot be used here. I have changed to simple array iteration to avoid using sets though.

File: kubetest2-gke/deployer/up.go (hunk 3 above)
Comment (chizhg) [reply kubetest2-gke/deployer/up.go]: Done.

File: kubetest2-gke/deployer/version_test.go [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+		{
+			desc:          "the same cluster version is a match",
+			version:       "1.19.10-gke.1000",
+			target:        "1.19.10-gke.1000",
+			expectMatched: true,
+		},
+		{
+			desc:          "the same major.minor version is a match",
+			version:       "1.19",
+			target:        "1.19.10-gke.1000",
```
Comment (amwat) [inline kubetest2-gke/deployer/version_test.go:143]: nit:

also add a case with `version` and `target` swapped to exercise 
```
if len(parts) < len(matchParts) {
		return false
	}
```

**Review by amwat — COMMENTED** [review]
/lgtm
/approve

/hold

couple of nits feel free to cancel the hold.

File: kubetest2-gke/deployer/version.go (hunk 2 above)
Comment (amwat) [reply kubetest2-gke/deployer/version.go]: I meant since the strings are the same, we can use `toLower`/`toUpper` and the default case to also use it as a validation method.
But considering it's just 4 values the array solution is fine as well.

File: kubetest2-gke/deployer/version_test.go (hunk 4 above)
Comment (chizhg) [reply kubetest2-gke/deployer/version_test.go:143]: 👍 I'll address this comment in my next PR.
